backend/homework/2/src/main/java/QuestionTwo/SentimentAnalyzer.java

```diff
@@ -0,0 +1,123 @@
+package QuestionTwo;
+
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+
+import java.util.Arrays;
+
+public class SentimentAnalyzer {
+    public static int[] detectProsAndCons(String review, String[][] featureSet, String[] posOpinionWords, String[] negOpinionWords) {
+        int[] featureOpinions = new int[featureSet.length];
+
+        for(int i = 0 ; i < featureSet.length ; i++){
+            String[] featureIndividual = featureSet[i];
+            for(int counterFeature = 0 ; counterFeature  < featureIndividual.length ; counterFeature++) {
+                boolean containsFeature = review.toLowerCase().contains(featureIndividual[counterFeature].toLowerCase());
+                if (!containsFeature) {
+                    continue;
+                } else {
+                    int getReview = getOpinionOnFeature(review.toLowerCase(), featureIndividual[counterFeature].toLowerCase(), posOpinionWords, negOpinionWords);
+                    featureOpinions[i] = getReview;
+                    break;
+                }
+            }
+        }
+        return featureOpinions;
+    }
+    private static int getOpinionOnFeature(String review, String feature,String[] posOpinionWords, String[] negOpinionWords) {
+        int ans = checkForWasPhrasePattern(review, feature, posOpinionWords, negOpinionWords);
+        if(ans == 0){
+            ans = checkForOpinionFirstPattern(review, feature, posOpinionWords, negOpinionWords);
+        }
+        return ans;
+    }
+    private static int checkForWasPhrasePattern(String review, String feature, String[] posOpinionWords, String[] negOpinionWords) {
+        int opinion = 0;
+        String pattern = feature + " was ";
+        if(review.contains(pattern)){
+            for(int counter = 0 ; counter < posOpinionWords.length ; counter++){
+                String anotherPattern = pattern + posOpinionWords[counter];
+                if(review.contains(anotherPattern)){
+                    opinion = 1;
+                    return opinion;
+                }else{
+                    continue;
+                }
+            }
+            for(int counter = 0 ; counter < negOpinionWords.length ; counter++){
```

Comment:
this and the above loop could be extracted into a different function and could be made generic to some extent to support both the cases

Reply:
noted